hedera-node/hedera-addressbook-service-impl/src/main/java/com/hedera/node/app/service/addressbook/impl/schemas/AddressBookTransplantSchema.java

```diff
@@ -44,15 +42,32 @@ default void restart(@NonNull final MigrationContext ctx) {
      * @param writableStates the state in which to store the node metadata
      */
     default int setNodeMetadata(@NonNull final Network network, @NonNull final WritableStates writableStates) {
+        requireNonNull(network);
+        requireNonNull(writableStates);
+
         final WritableKVState<EntityNumber, Node> nodes = writableStates.get(NODES_KEY);
         final var adoptedNodeCount = new AtomicInteger();
+        final var liveNodeIds = new HashSet<EntityNumber>();
         network.nodeMetadata().stream()
                 .filter(NodeMetadata::hasNode)
                 .map(NodeMetadata::nodeOrThrow)
                 .forEach(node -> {
+                    final var id = new EntityNumber(node.nodeId());
                     adoptedNodeCount.getAndIncrement();
-                    nodes.put(new EntityNumber(node.nodeId()), node);
+                    liveNodeIds.add(id);
+                    nodes.put(id, node);
                 });
+        log.info("Adopted nodes {} from startup assets (override-network.json)", liveNodeIds);
+
+        nodes.keys().forEachRemaining(nodeId -> {
+            final var node = requireNonNull(nodes.get(nodeId));
+            if (!liveNodeIds.contains(nodeId)) {
+                nodes.put(
+                        new EntityNumber(nodeId.number()),
+                        node.copyBuilder().deleted(true).build());
```

Comment:
The ticket states that an override network should _completely_ replace the existing address book. If an existing node isn't in the override file, should we be removing it entirely (instead of only marking it as deleted)? A future nodeUpdate transaction could bring the deleted node back to life, which seems contrary to the idea of complete replacement

Reply:
We don't remove any node from state currently. It can be done safely only after this ticket is complete https://github.com/hiero-ledger/hiero-consensus-node/issues/18855.